kernel/src/scan/mod.rs

```diff
@@ -1490,4 +1478,182 @@ mod tests {
         );
         Ok(())
     }
+
+    #[test]
+    fn test_state_info_no_partition_columns() {
```

Comment:
thanks for the extra tests. can we also make a good-first-issue follow-up to move the scan/mod.rs tests to their own file?